quinn-proto/src/connection/streams.rs

```diff
@@ -1034,6 +1048,27 @@ impl Recv {
         }
     }
 
+    /// Returns the window that should be advertised for this Stream in a
```

Comment:
I think I'd prefer to leave the `announcable` prefixes and infixes out here, that feels a little wordy and doesn't seem to add that much value.

Also in doc comments, probably should keep the first line of the doc comment a bit shorter so that it fits on a line, and our convention is to not end these with a period (and have an empty line between the first one and the second, perhaps?).

Reply:
I updated to follow the  comment conventions, and removed some of the `announcable` in the body.

I wouldn't like to remove it from the function name however, since the destinction seems important here.

Reply:
Distinction between what and what? I think the `update` serves to make it clear enough.

Reply:
Distinction between:
- The maximum window we could announce at all.
- A window update we could sent, but won't do it due to it not being deemed significant enough

I guess I can also change it by switching the return type of the function to `fn window_update(&self, stream_receive_window: u64) ->(u64, bool)` where the bool indicates whether transmission is wanted. Or call it `max_stream_data`.  Will leave it to you which option is preferred, and make that change

Reply:
To me it makes sense that the stream gets to decide whether a window update needs to be announced. If the stream decides it needs to be announced, it will return it to the connection, which will just act on the stream's behalf. As such, between the connection and stream there's no necessity to introduce the concept of announceability. (And therefore I prefer the original `Option`-based return type.)